### PR TITLE
Add a PR template

### PR DESCRIPTION
**Pull request by julien-deramond**
This PR proposes to add a PR template to avoid:
* having to copy/paste [DoD for feature](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/DoD-for-feature) or [DoD for release](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/DoD-for-release)
* forgetting to do so

Moreover it will help the contributors to provide the needed information.

This is based on https://github.com/twbs/bootstrap/pull/35079.

You can test it by creating a new pull request in https://github.com/julien-deramond/github-repository-test/pulls (use the _main-jd-fake-branch_).

## After the merge

- [ ] Remove [DoD for feature](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/DoD-for-feature)
- [ ] Remove [DoD for release](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/DoD-for-release)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+### Description
+
+<!-- Describe your changes in detail -->
+
+### Motivation & Context
```
Comment (louismaximepiton) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: I'd regroup this part with the `Related Issues`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  - [ ] when bumping minor version: ensure `dist` URLs in examples' HTML has changed
+  - [ ] double-check everything before pushing, starting by searching for forgotten old version number occurences
+- [ ] make an announcement in Plazza :tada:
+-->
+
+### Related issues
+
+<!-- Please link any related issues here. -->
+
+### [Live preview](https://deploy-preview-{your pr number}--boosted.netlify.app/)
```
Comment (louismaximepiton) [inline .github/PULL_REQUEST_TEMPLATE.md:130]: I'd move this part to the top

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  - in /forms/overview/ if it is a new Orange custom component that is a form control
+- [ ] Check (and fix) RTL version
+- [ ] Run linters
+- [ ] Run compilers
+- [ ] Tests added for JS-side
+- [ ] Run tests
+- [ ] Manually run BrowserStack test
+- [ ] Manually run Percy test
+- [ ] Cross-browser test:
+  - Firefox ESR
```
Comment (louismaximepiton) [inline .github/PULL_REQUEST_TEMPLATE.md:51]: Maybe add `[ ]` :
- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari

Comment (MewenLeHo) [reply .github/PULL_REQUEST_TEMPLATE.md:51]: We can also add:

- [ ] Desktop
- [ ] Small screens

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Should match specs (eg. either the Web UI Kit or any pattern from the WAI — or both…)
+- [ ] Docs added:
+  - including the "Sass" part using `scss-docs` shortcode
+  - in /about/overview/#custom-components if it is a new Orange custom component
+  - in /getting-started/introduction/#components if it is a new Orange custom component that requires JavaScript (and Popper)
+  - in /customize/overview#csps-and-embedded-svgs if it is a new Orange custom component that includes embedded SVGs in our CSS
+  - in /forms/validation/?#supported-elements if it is a new Orange custom component that is a form control
+  - in /forms/overview/ if it is a new Orange custom component that is a form control
+- [ ] Check (and fix) RTL version
+- [ ] Run linters
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:44]: Maybe remind the commands to run by adding :  _(using `npm run ..., npm run ..., etc.`)_

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Run compilers
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:45]: Maybe remind the commands to run by adding :  _(using `npm run ..., npm run ..., etc.`)_

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Tests added for JS-side
+- [ ] Run tests
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:47]: Maybe remind the commands to run by adding : (using `npm run test`)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+### Reviews
+
+- [ ] Code review
+- [ ] A11y review
+- [ ] Design review
+-->
+
+<!-- /!\ Core Team only: Uncomment the following for a release DoD -->
+<!--
+- [ ] Run linters;
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:68]: Maybe remind the commands to run by adding : _(using `npm run ..., npm run ..., etc.`)_

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Run compilers;
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:69]: Maybe remind the commands to run by adding : _(using `npm run ..., npm run ..., etc.`)_

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Run tests;
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:70]: Maybe remind the commands to run by adding : _(using `npm run test`)_

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Check documentation site: examples and contents;
+- [ ] Test cross-browser compatibility locally and with [BrowserStack](https://www.browserstack.com/):
+  - Firefox ESR
+  - IE11 (v4 only)
+  - Latest Edge, Chrome, Firefox, Safari
+  - iOS Safari
+  - Chrome & Firefox on Android
```
Comment (isabellechanclou) [inline .github/PULL_REQUEST_TEMPLATE.md:77]: Have here the same list as above (if made suggestions are accepted)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (isabellechanclou) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: Or move `Related Issues` up here or just before this paragraph ?

**Comment by julien-deramond** [comment]
I modified a bit the file architecture to gather what's been discussed in Bootstrap and here (/cc @louismaximepiton @MewenLeHo @isabellechanclou ).
I haven't added the corresponding `npm run` commands within the template for now. It could be done in a second PR after having tried this first version.
What do you think all? Do we start with that?

**Comment by julien-deramond** [comment]
> An idea just popped up to my mind, should we add a checklist for reviewers? I mean something to be sure that we aren't missing something obvious as reviewer? It doesn't prevent to check some other things that were not noticed, and could be filled up by other reviewers. Otherwise the actual version should do the work.

IMO the checklist for the reviewer is the same as the one for the person proposing the PR; you got to be your own reviewer when you propose your PR.

**Comment by MewenLeHo** [comment]
> An idea just popped up to my mind, should we add a checklist for reviewers? I mean something to be sure that we aren't missing something obvious as reviewer? It doesn't prevent to check some other things that were not noticed, and could be filled up by other reviewers. Otherwise the actual version should do the work.

Should we do it? Maybe.
Can we do it? I honestly don't know. Most PRs are really different and about various subject. Would it be possible to find a common set of points to check?

**Comment by julien-deramond** [comment]
> Should we do it? Maybe. Can we do it? I honestly don't know. Most PRs are really different and about various subject. Would it be possible to find a common set of points to check?

Most of the checks are common sense:
- The PR corresponds to what's in the issue (if issue)
- The PR doesn't break anything
- Check all the possible non-regressions (not only the use case mentioned in the description)
- Think about the users, migrations etc.
- Write tests

IMO the common part is rather obvious and the rest is, as @MewenLeHo said, "really different and about various subject" and depending on the context.

**Comment by louismaximepiton** [comment]
Let's merge it this way! As discussed with @julien-deramond, could come later but could be useful for external contributors or newcomers to have at least a base for their own review.

**Pull request merged**